A-vs-An/AvsAn-Python/AvsAn.py

```diff
@@ -0,0 +1,56 @@
+import json
+import os
+
+class AvsAn():
+    """
+    Singleton class, AvsAn.
+    """
+
+    __instance = None
+
+    @staticmethod
+    def getInstance():
+        """ Static access method. """
+        if AvsAn.__instance == None:
+            AvsAn()
+        return AvsAn.__instance
+
+    def __init__(self):
+        """ Virtually private constructor. """
+        if AvsAn.__instance != None:
+            raise Exception("This class is a singleton!")
+        else:
+            __location__ = os.path.realpath(os.path.join(os.getcwd(), os.path.dirname(__file__)))
+            with open(os.path.join(__location__, 'a_vs_an.json')) as f:
```

Comment:
In the other implementations this kind of complexity is avoided by embedding the actual data in the source itself.  Is that hard to do in python?  Alternatively, you'll want to have the path of the *python source* as base, not cwd, since a consumer might reasonably be expected to run such a script from another location or to even to change the working directory.

Reply:
Basically: if you took the contents of a_vs_an.json, replaced `\` by `\\` and `'` by `\'`, wrapped it in single quote `'` chars: would that be a valid python string?

Reply:
I'm not at all fluent in python anymore, but `os.getcwd()` sounds like it's getting the current working directory - not the directory of this source file.

I do agree is "cleaner" to keep the data separate, but that's largely only ever an advantage to the updater/maintainer, not the end-user.  A python programmer using this as a library would probably rather it just work without any file-system-layout dependencies than care too much whether the dictionary is easily updatable.

For what it's worth: I have at times rerun the analysis to update the dictionary, but the updates are truly minor - as is to be expected, since language doesn't evolve *that* quickly and even if there are new words this model has never seen before - they'll often follow patterns of words that *have* been seen and thus the longest-matching-prefix approach still works OK.